src/MysteriousEnd.js

```diff
@@ -0,0 +1,16 @@
+
+// we modify the prototype of Array creation and define extra function
+Array.prototype.end=(function(){
+    return function() {
+        return this[this.length - 1]
+    };
+})();
+
+
+// this function iterates array keys and since its prototype has an extra value defined, it will iterate it as well
+for (var i in [1, 2, 3, 4]) {console.log(i)}
```

Comment:
I really do not believe this will print:
0
1
2
3
end

Reply:
ran it both in browser (chrome) and my node.js 8.4, output is exactly like that. Did you get something else?

Reply:
You convinced me, it really works. Now i don't believe that the function is necessary 😄

Reply:
Yes, it works even when you just set it to random value like this:
Array.prototype.end=1

Reply:
My favourite is Array.prototype.end = undefined